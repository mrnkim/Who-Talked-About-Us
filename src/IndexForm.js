import { React, useState, useEffect } from "react";
import InputForm from "./InputForm";
import TwelveLabsApi from "./api";
import { Alert } from "react-bootstrap";

function IndexForm({ indexes, setIndexes, addIndex }) {
  const [indexName, setIndexName] = useState("");
  const [error, setError] = useState("");

  useEffect(() => {
    console.log(indexes);
  }, [indexes]);

  /** Update form input. */
  function handleChange(evt) {
    const input = evt.target;
    setIndexName(input.value);
  }

  /** Call parent function and clear form. */
  async function handleSubmit(evt) {
    evt.preventDefault();
    const trimmedIndexName = indexName.trim();
    const newIndex = await addIndex(trimmedIndexName);
  }

  return (
    <div>
      <InputForm
        handleSubmit={handleSubmit}
        handleChange={handleChange}
        input={indexName}
        type="indexName"
        buttonText="Create Index"
      />

      {error && (
        <Alert variant="danger" dismissible>
          {error}
        </Alert>
      )}
    </div>
  );
}

export default IndexForm;
