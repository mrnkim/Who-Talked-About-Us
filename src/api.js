import axios from "axios";

const API_URL = process.env.REACT_APP_API_URL;
const API_KEY = process.env.REACT_APP_API_KEY;

class TwelveLabsApi {
  // Declare the `headers` object containing your API key
  static headers = {
    "x-api-key": API_KEY,
  };

  static async getIndexes() {
    const config = {
      method: "GET",
      url: `${API_URL}/indexes`,
      headers: this.headers,
    };

    try {
      const response = await axios.request(config);
      return response.data.data;
    } catch (error) {
      console.error(error);
    }
  }

  static async createIndex(indexName) {
    const INDEXES_URL = `${API_URL}/indexes`;
    const INDEX_NAME = indexName;

    // Make a POST request to the `indexes` endpoint, passing the `x-api-key` header parameter
    try {
      const resp = await axios.post(
        INDEXES_URL,
        {
          engine_id: "marengo2.5",
          index_options: ["visual", "conversation", "text_in_video", "logo"],
          index_name: INDEX_NAME,
        },
        { headers: this.headers }
      );
      const { data: response } = resp;
      return resp.data;
    } catch (error) {
      console.error(`Error: ${error}`);
    }
  }

  static async getVideos(index_id) {
    const config = {
      method: "GET",
      url: `${API_URL}/indexes/${index_id}/videos`,
      headers: this.headers,
    };

    try {
      const response = await axios.request(config);
      return response.data.data;
    } catch (error) {
      console.error(error);
    }
  }

  static async getVideo(index_id, video_id) {
    const config = {
      method: "GET",
      url: `${API_URL}/indexes/${index_id}/videos/${video_id}`,
      headers: {
        ...this.headers,
        "Content-Type":
          "multipart/form-data; boundary=---011000010111000001101001",
      },
    };

    try {
      const response = await axios.request(config);
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }

  static async searchVideo(indexId, query) {
    const config = {
      method: "POST",
      url: `${API_URL}/search`,
      headers: {
        ...this.headers,
        " accept": "application/json",
        "Content-Type": "application/json",
      },
      data: {
        index_id: `${indexId}`,
        search_options: ["visual", "conversation", "text_in_video", "logo"],
        query: { text: `${query}` },
      },
    };

    try {
      const response = await axios.request(config);
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }
  static async checkStatus(taskId) {
    const config = {
      method: "GET",
      url: `${API_URL}/tasks/${taskId}`,
      headers: {
        ...this.headers,
      },
    };

    try {
      const response = await axios.request(config);
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }

  static async uploadVideo(indexId, videoUrl) {
    try {
      let formData = new FormData();
      formData.append("index_id", indexId);
      formData.append("language", "en");
      formData.append("video_url", videoUrl);

      let config = {
        method: "post",
        url: `${API_URL}/tasks`,
        headers: {
          ...this.headers,
          "Content-Type": "multipart/form-data",
        },
        data: formData,
      };

      let resp = await axios(config);
      let response = await resp.data;
      const VIDEO_ID = response.video_id;
      return response;
    } catch (error) {
      console.error(`Error: ${error}`);
      console.error(`Error response: ${error.response.data}`);
    }
  }
}

export default TwelveLabsApi;
