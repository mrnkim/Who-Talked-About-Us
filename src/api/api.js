import axios from "axios";

const API_URL = process.env.REACT_APP_API_URL;
const API_KEY = process.env.REACT_APP_API_KEY;

/** API Class
 *
 * Static class tying together methods used to get/send to to the API.
 *
 */
class TwelveLabsApi {
  /** Declare the `headers` object containing your API key */
  static headers = {
    "x-api-key": API_KEY,
  };

  /** Get indexes of a user */
  static async getIndexes() {
    const config = {
      method: "GET",
      url: `${API_URL}/indexes`,
      headers: this.headers,
    };

    try {
      const response = await axios.request(config);
      return response.data.data;
    } catch (error) {
      console.error(error);
    }
  }

  /** Creates an index */
  static async createIndex(indexName) {
    const INDEXES_URL = `${API_URL}/indexes`;
    const INDEX_NAME = indexName;

    try {
      const resp = await axios.post(
        INDEXES_URL,
        {
          engine_id: "marengo2.5",
          index_options: ["visual", "conversation", "text_in_video", "logo"],
          index_name: INDEX_NAME,
        },
        { headers: this.headers }
      );
      const { data: response } = resp;
      return resp.data;
    } catch (error) {
      console.error(`Error: ${error}`);
    }
  }

  /** Deletes an index */
  static async deleteIndex(indexId) {
    const config = {
      method: "DELETE",
      url: `${API_URL}/indexes/${indexId}`,
      headers: this.headers,
    };
    try {
      const response = await axios.request(config);
      return response;
    } catch (error) {
      console.error(error);
    }
  }

  /** Get all videos of an index */
  static async getVideos(index_id) {
    const config = {
      method: "GET",
      params: { page_limit: "50" },
      url: `${API_URL}/indexes/${index_id}/videos`,
      headers: this.headers,
    };

    try {
      const response = await axios.request(config);
      return response.data.data;
    } catch (error) {
      console.error(error);
    }
  }

  /** Get details on a video */
  static async getVideo(index_id, video_id) {
    const config = {
      method: "GET",
      url: `${API_URL}/indexes/${index_id}/videos/${video_id}`,
      headers: {
        ...this.headers,
        "Content-Type":
          "multipart/form-data; boundary=---011000010111000001101001",
      },
    };

    try {
      const response = await axios.request(config);
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }

  /** Search videos with a given query */
  static async searchVideo(indexId, query) {
    const config = {
      method: "POST",
      url: `${API_URL}/search`,
      headers: {
        ...this.headers,
        " accept": "application/json",
        "Content-Type": "application/json",
      },
      data: {
        index_id: `${indexId}`,
        search_options: ["visual", "conversation", "text_in_video", "logo"],
        query: { text: `${query}` },
      },
    };

    try {
      const response = await axios.request(config);
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }

  /** Checks status of a task (e.g., video upload task) */
  static async checkStatus(taskId) {
    const config = {
      method: "GET",
      url: `${API_URL}/tasks/${taskId}`,
      headers: {
        ...this.headers,
      },
    };

    try {
      const response = await axios.request(config);
      return response.data;
    } catch (error) {
      console.error(error);
    }
  }

  /** Uploads a video with a given url */
  static async uploadVideo(indexId, videoUrl) {
    try {
      let formData = new FormData();
      formData.append("index_id", indexId);
      formData.append("language", "en");
      formData.append("video_url", videoUrl);

      let config = {
        method: "post",
        url: `${API_URL}/tasks`,
        headers: {
          ...this.headers,
          "Content-Type": "multipart/form-data",
        },
        data: formData,
      };

      let resp = await axios(config);
      let response = await resp.data;
      return response;
    } catch (error) {
      console.error(`Error: ${error}`);
      console.error(`Error response: ${error.response.data}`);
    }
  }

  /** Deletes a video */
  static async deleteVideo(indexId, videoId) {
    const config = {
      method: "DELETE",
      url: `${API_URL}/indexes/${indexId}/videos/${videoId}`,
      headers: this.headers,
    };
    try {
      const response = await axios.request(config);
      return response.status;
    } catch (error) {
      console.error(error);
    }
  }
}

export default TwelveLabsApi;
