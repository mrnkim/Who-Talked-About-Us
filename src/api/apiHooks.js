import {
  useQuery,
  useQueries,
  useMutation,
  useQueryClient,
} from "@tanstack/react-query";
import { keys } from "./keys";

const SERVER_BASE_URL = new URL(
  `${process.env.REACT_APP_SERVER_URL}:${process.env.REACT_APP_PORT_NUMBER}`
);
const INDEXES_URL = new URL("/indexes", SERVER_BASE_URL);
const INDEX_URL = new URL("/index", SERVER_BASE_URL);
const SEARCH_URL = new URL("/search", SERVER_BASE_URL);
const TASKS_URL = new URL("/tasks", SERVER_BASE_URL);

export function useGetIndexes(page, pageLimit) {
  return useQuery({
    queryKey: [keys.INDEXES, page],
    queryFn: () =>
      fetch(`${INDEXES_URL}?page=${page}&page_limit=${pageLimit}`).then((res) =>
        res.json()
      ),
  });
}

export function useGetIndex(indexId) {
  return useQuery({
    queryKey: [keys.INDEX],
    queryFn: () => fetch(`${INDEXES_URL}/${indexId}`).then((res) => res.json()),
  });
}

export function useCreateIndex() {
  const queryClient = useQueryClient();
  return useMutation({
    mutationFn: (indexName) =>
      fetch(`${INDEXES_URL}`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ indexName }),
      }).then((res) => res.json()),
    onSuccess: () => queryClient.invalidateQueries([keys.INDEXES]),
    mutationKey: "createIndex",
  });
}

export function useDeleteIndex() {
  const queryClient = useQueryClient();
  return useMutation({
    mutationFn: (indexId) =>
      fetch(`${INDEXES_URL}?indexId=${indexId}`, {
        method: "DELETE",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ indexId }),
      }).then((res) => res.json()),
    onSuccess: () => queryClient.invalidateQueries([keys.INDEXES]),
    mutationKey: "deleteIndex",
  });
}

export function useGetVideos(currIndex, vidPage, vidPageLimit) {
  return useQuery({
    queryKey: [keys.VIDEOS, currIndex, vidPage],
    queryFn: () =>
      fetch(
        `${INDEXES_URL}/${currIndex}/videos?page=${vidPage}&page_limit=${vidPageLimit}`
      ).then((res) => res.json()),
  });
}

export function useGetAllAuthors(indexId) {
  return useQuery({
    queryKey: [keys.AUTHORS, indexId],
    queryFn: () =>
      fetch(`${INDEXES_URL}/${indexId}/authors`).then((res) => res.json()),
  });
}

export function useSearchVideo(indexId, query) {
  return useQuery({
    queryKey: [keys.SEARCH, indexId, query],
    queryFn: () => {
      return fetch(`${SEARCH_URL}`, {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ indexId, query }),
      }).then((res) => res.json());
    },
  });
}

export function useGetTask(taskId) {
  return useQuery({
    queryKey: [keys.TASK, taskId],
    queryFn: () => fetch(`${TASKS_URL}/${taskId}`).then((res) => res.json()),
    refetchInterval: (data) => {
      return data?.status === "ready" ? false : 5000;
    },
    refetchIntervalInBackground: true,
  });
}

export function useGetVideoOfSearchResults(indexId, query) {
  const { data: searchVideoQuery, refetch } = useSearchVideo(indexId, query);

  const searchResults = searchVideoQuery.data || [];

  const results = useQueries({
    queries: searchResults.map((result) => ({
      queryKey: [keys.SEARCH, indexId, result.video_id],
      queryFn: () =>
        fetch(`${INDEXES_URL}/${indexId}/videos/${result.video_id}`).then(
          (res) => res.json()
        ),
    })),
  });

  // Extract 'data' property from each object in 'results'
  const searchResultVideos = results.map(({ data }) => data);

  return { searchResults, searchResultVideos, refetch };
}
