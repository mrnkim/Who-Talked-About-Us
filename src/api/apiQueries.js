import { useQuery, useMutation } from "@tanstack/react-query";
import TwelveLabsApi from "./api";

export function useGetIndexes() {
  return useQuery({ key: ["indexes"], queryFn: TwelveLabsApi.getIndexes });
}

export function useCreateIndex() {
  return useMutation({
    mutationFn: (indexName) => TwelveLabsApi.createIndex(indexName),
  });
}
