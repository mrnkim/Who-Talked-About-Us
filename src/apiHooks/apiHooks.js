import {
  useQuery,
  useQueries,
  useMutation,
  useQueryClient,
  useInfiniteQuery,
} from "@tanstack/react-query";
import keys from "./keys";
import apiConfig from "./apiConfig";

export function useGetIndexes(page, pageLimit) {
  return useQuery({
    queryKey: [keys.INDEXES, page],
    queryFn: () =>
      apiConfig.TWELVE_LABS_API.get(
        `${apiConfig.INDEXES_URL}?page=${page}&page_limit=${pageLimit}`
      ).then((res) => res.data),
  });
}

export function useGetIndex(indexId) {
  return useQuery({
    queryKey: [keys.INDEX, indexId],
    queryFn: async () => {
      const response = await apiConfig.TWELVE_LABS_API.get(
        `${apiConfig.INDEXES_URL}/${indexId}`
      );
      if (response.data.error) {
        return { error: response.data.error };
      }
      return response.data;
    },
  });
}

export function useCreateIndex(setIndexId) {
  const queryClient = useQueryClient();
  return useMutation({
    mutationFn: (indexName) =>
      apiConfig.TWELVE_LABS_API.post(apiConfig.INDEXES_URL, { indexName }).then(
        (res) => res.data
      ),
    onSuccess: (newIndex) => {
      setIndexId(newIndex._id);
      queryClient.invalidateQueries([keys.INDEX, newIndex._id]);
    },
    mutationKey: "createIndex",
  });
}

export function useDeleteIndex(setIndexId) {
  return useMutation({
    mutationFn: (indexId) =>
      apiConfig.TWELVE_LABS_API.delete(
        `${apiConfig.INDEXES_URL}?indexId=${indexId}`
      ).then((res) => res.data),
    onSuccess: () => {
      setIndexId(null);
    },
    mutationKey: "deleteIndex",
  });
}

export function useGetVideos(indexId, page, pageLimit) {
  return useQuery({
    queryKey: [keys.VIDEOS, indexId, page],
    queryFn: async () => {
      try {
        const response = await apiConfig.TWELVE_LABS_API.get(
          `${apiConfig.INDEXES_URL}/${indexId}/videos`,
          {
            params: { page, page_limit: pageLimit },
          }
        );
        return response.data;
      } catch (error) {
        throw error.response?.data || error;
      }
    },
  });
}

export function useGetAllAuthors(indexId) {
  return useQuery({
    queryKey: [keys.AUTHORS, indexId],
    queryFn: () =>
      apiConfig.TWELVE_LABS_API.get(
        `${apiConfig.INDEXES_URL}/${indexId}/authors`
      ).then((res) => res.data),
  });
}

export function useSearchVideo(indexId, query) {
  return useQuery({
    queryKey: [keys.SEARCH, indexId, query],
    queryFn: () =>
      apiConfig.TWELVE_LABS_API.post(apiConfig.SEARCH_URL, {
        indexId,
        query,
      }).then((res) => res.data),
  });
}

export function useGetSearchResults(indexId, pageToken) {
  return useInfiniteQuery({
    queryKey: [keys.SEARCH, pageToken],
    queryFn: () =>
      apiConfig.TWELVE_LABS_API.get(
        `${apiConfig.SEARCH_URL}/${pageToken}`
      ).then(async (res) => {
        const searchData = res.data;

        // Fetch videos for each search result
        const videosPromises = searchData.data.map(async (searchResult) => {
          const videoResponse = await apiConfig.TWELVE_LABS_API.get(
            `${apiConfig.INDEXES_URL}/${indexId}/videos/${searchResult.id}`
          );
          return videoResponse.data;
        });

        // Wait for all video requests to complete
        const videos = await Promise.all(videosPromises);

        // Attach videos to the search result data
        const searchDataWithVideos = {
          ...searchData,
          videos: videos,
        };
        return searchDataWithVideos;
      }),
    getNextPageParam: (lastPage) => {
      const nextPageToken = lastPage.page_info.next_page_token || undefined;
      return nextPageToken || null;
    },
  });
}

export function useGetTask(taskId) {
  return useQuery({
    queryKey: [keys.TASK, taskId],
    queryFn: () =>
      apiConfig.TWELVE_LABS_API.get(`${apiConfig.TASKS_URL}/${taskId}`).then(
        (res) => res.data
      ),
    refetchInterval: (data) => {
      return data?.status === "ready" ? false : 5000;
    },
    refetchIntervalInBackground: true,
  });
}

export function useGetVideosOfSearchResults(indexId, query) {
  const {
    data: initialSearchData,
    refetch,
    isLoading,
  } = useSearchVideo(indexId, query);
  const initialSearchResults = initialSearchData.data || [];

  const resultVideos = useQueries({
    queries: initialSearchResults.map((searchResult) => ({
      queryKey: [keys.SEARCH, indexId, searchResult.id],
      queryFn: () =>
        apiConfig.TWELVE_LABS_API.get(
          `${apiConfig.INDEXES_URL}/${indexId}/videos/${searchResult.id}`
        ).then((res) => res.data),
    })),
  });
  const initialSearchResultVideos = resultVideos.map(({ data }) => data);
  return {
    initialSearchData,
    initialSearchResults,
    initialSearchResultVideos,
    refetch,
    isLoading,
  };
}

export async function fetchNextPageSearchResults(queryClient, nextPageToken) {
  try {
    const response = await queryClient.fetchQuery({
      queryKey: [keys.SEARCH, nextPageToken],
      queryFn: async () => {
        const response = await apiConfig.TWELVE_LABS_API.get(
          `${apiConfig.SEARCH_URL}/${nextPageToken}`
        );
        const data = response.data;
        return data;
      },
    });
    return response;
  } catch (error) {
    console.error("Error fetching next page of search results:", error);
    throw error;
  }
}

export async function fetchNextPageSearchResultVideos(
  queryClient,
  currIndex,
  nextPageResultId
) {
  try {
    const response = await queryClient.fetchQuery({
      queryKey: [keys.VIDEOS, currIndex, nextPageResultId],
      queryFn: async () => {
        const response = await apiConfig.TWELVE_LABS_API.get(
          `${apiConfig.INDEXES_URL}/${currIndex}/videos/${nextPageResultId}`
        );
        const data = response.data;
        return data;
      },
    });
    return response;
  } catch (error) {
    console.error("Error fetching next page of search results:", error);
    throw error;
  }
}
