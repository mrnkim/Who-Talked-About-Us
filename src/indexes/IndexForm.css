.errorMessage {
    color: var(--TL-Red-500, #F44336);
    /* Body2 */
    font-size: 0.875rem;
    font-style: normal;
    font-weight: 400;
    line-height: 1.25rem;
}

.inputAndError{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 0.25rem;
}