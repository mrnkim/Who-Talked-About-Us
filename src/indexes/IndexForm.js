import React, { useState } from "react";
import { useContext } from "react";
import setIndexIdContext from "../common/setIndexIdContext";
import InputForm from "../common/InputForm";
import icon from "../svg/Create.svg";
import "./IndexForm.css";
import { useCreateIndex } from "../apiHooks/apiHooks";

/** Renders the input form for an index
 *
 * App -> IndexForm -> InputForm
 *
 */

function IndexForm() {
  const [indexName, setIndexName] = useState("");
  const [error, setError] = useState("");
  const { setIndexId } = useContext(setIndexIdContext);

  const createIndexMutation = useCreateIndex(setIndexId);

  function handleChange(evt) {
    const input = evt.target;
    setIndexName(input.value);
    if (error && input.value.trim() !== "") {
      setError("");
    }
  }

  async function handleSubmit(evt) {
    evt.preventDefault();
    const trimmedIndexName = indexName.trim();
    if (!trimmedIndexName) {
      setError("Please enter the name of an index");
    } else {
      try {
        const { error } = await createIndexMutation.mutateAsync(
          trimmedIndexName
        );
        if (error) {
          setError(
            error.status === 409
              ? `Index name ${trimmedIndexName} already exists. Please use another unique name and try again.`
              : "Error creating index. Please try again."
          );
        } else {
          setIndexName("");
        }
      } catch (error) {
        setError("An error occurred. Please try again later.");
      }
    }
  }

  return (
    <div className="inputAndError">
      <InputForm
        handleSubmit={handleSubmit}
        handleChange={handleChange}
        value={indexName}
        type="Create Index (e.g., January Videos)"
        icon={icon}
        className="textField"
      />
      {error && <div className="errorMessage">{error}</div>}
    </div>
  );
}

export default IndexForm;
