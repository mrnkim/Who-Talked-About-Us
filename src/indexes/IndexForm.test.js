import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import IndexForm from "./IndexForm";
import { useCreateIndex } from "../api/apiHooks";

// Mock the useCreateIndex hook
jest.mock("../api/apiHooks", () => ({
  useCreateIndex: () => ({
    mutateAsync: jest.fn(),
  }),
}));

describe("IndexForm", () => {
  let queryClient;

  beforeEach(() => {
    queryClient = new QueryClient();
  });

  it("renders without crashing", () => {
    render(
      <QueryClientProvider client={queryClient}>
        <IndexForm />
      </QueryClientProvider>
    );

    expect(
      screen.getByLabelText(/Create an index and start analyzing videos!/i)
    ).toBeInTheDocument();
  });
});
