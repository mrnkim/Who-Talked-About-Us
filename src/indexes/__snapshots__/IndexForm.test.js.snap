// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div
      class="mt-5"
    >
      <div
        class="row"
        style="display: flex; justify-content: center;"
      >
        <div
          class="col-lg-8 col-md-8 col-sm-12"
        >
          <form
            class="d-flex"
          >
            <input
              aria-label="Enter an index name"
              class="form-control"
              data-testid="search-input"
              id="Enter an index name"
              name="Enter an index name"
              placeholder="Enter an index name"
              style="width: 60%; font-size: 1em;"
              type="Enter an index name"
              value=""
            />
            <button
              class="rounded btn btn-primary"
              style="margin-left: 3px;"
              type="submit"
            >
              Create Index
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
